workflows/gpu_operator_dashboard/fetch_ci_data.py

```
@@ -88,22 +89,41 @@ class TestResult:
+        repo, pr_number, job_name, build_id = extract_build_components(self.prow_job_url)
+        return (pr_number, job_name, build_id)
+
+    def has_exact_versions(self) -> bool:
+        """Check if this result has exact semantic versions (not base versions from URL)."""
+        try:
+            """Exact versions parse as semver (e.g., 4.18.23, 24.9.2), not bases like 4.18 or 24-9-x."""
+            ocp = self.ocp_full_version
+            gpu = self.gpu_operator_version.split("(")[0].strip()
+            semver.VersionInfo.parse(ocp)
+            semver.VersionInfo.parse(gpu)
+            return True
```

Ruff (0.12.2)

Consider moving this statement to an `else` block

(TRY300)